source/module_hamilt_pw/hamilt_pwdft/stress_func_exx.cpp

```diff
@@ -127,25 +156,73 @@ void Stress_PW<FPTYPE, Device>::stress_exx(ModuleBase::matrix& sigma,
     // prepare for the potential
     for (int ik = 0; ik < nks; ik++)
     {
-        for (int iq = 0; iq < nqs; iq++)
+        for (int iq = 0; iq < nks; iq++)
         {
-            auto k = wfcpw->kvec_c[ik];
-            auto q = wfcpw->kvec_c[iq];
+            auto k_c = wfcpw->kvec_c[ik];
+            auto k_d = wfcpw->kvec_d[ik];
+            auto q_c = wfcpw->kvec_c[iq];
+            auto q_d = wfcpw->kvec_d[iq];
+
             #ifdef _OPENMP
             #pragma omp parallel for schedule(static)
             #endif
             for (int ig = 0; ig < rhopw->npw; ig++)
             {
-                FPTYPE qq = (k - q + rhopw->gcar[ig]).norm2() * tpiba2;
-                FPTYPE fac = -ModuleBase::FOUR_PI * ModuleBase::e2 / qq;
-                if (qq < 1e-8)
+                auto g_d = rhopw->gdirect[ig];
+                auto kqg_d = k_d - q_d + g_d;
+                Real grid_factor = 1;
+                if (gamma_extrapolation)
                 {
-                    pot[ig + iq * rhopw->npw + ik * rhopw->npw * nqs] = exx_div;
+                    if (isint(kqg_d[0] * nqs_half1) && isint(kqg_d[1] * nqs_half2) && isint(kqg_d[2] * nqs_half3))
+                    {
+                        grid_factor = 0;
+                    }
+                    else
+                    {
+                        grid_factor = 8.0/7.0;
+                    }
                 }
+
+                Real gg = (k_c - q_c + rhopw->gcar[ig]).norm2() * tpiba2;
+                Real hse_omega2 = GlobalC::exx_info.info_global.hse_omega * GlobalC::exx_info.info_global.hse_omega;
+                // if (kqgcar2 > 1e-12) // vasp uses 1/40 of the smallest (k spacing)**2
+                if (gg >= 1e-8)
+                {
+                    Real fac = -ModuleBase::FOUR_PI * ModuleBase::e2 / gg;
+                    // if (PARAM.inp.dft_functional == "hse")
+                    if (GlobalC::exx_info.info_global.ccp_type == Conv_Coulomb_Pot_K::Ccp_Type::Erfc)
+                    {
+                        pot[ik * nks * rhopw->npw + iq * rhopw->npw + ig] = fac * (1.0 - std::exp(-gg / 4.0 / hse_omega2)) * grid_factor;
```

Comment:
const int index = ik * nks * rhopw->npw + iq * rhopw->npw + ig; and then use index